static/src/javascripts/projects/common/modules/identity/upsell/account-creation/AccountCreationFlow.js

```diff
@@ -0,0 +1,209 @@
+    onAccountCreated: () => {},
+};
+
+type AccountCreationFeaturesProps = {
+    returnUrl: string,
+};
+
+class AccountCreationFeatures extends Component<AccountCreationFeaturesProps> {
+    render() {
+        return (
+            <div>
+                <hr className="manage-account-small-divider" />
+                <h1 className="identity-title--small">
+                    Thanks for creating a Guardian account!
+                </h1>
+                <div className="identity-forms-message__body">
+                    <p>
+                        You are now signed in. Start exploring your benefits
+                        from our home page.
+                    </p>
+                </div>
+                <div className="identity-forms-message__body">
+                    <a
+                        className="manage-account__button manage-account__button--icon manage-account__button--main"
+                        data-link-name="complete-consents : cta-bottom"
+                        href={this.props.returnUrl}>
+                        Go to The Guardian
+                        <span
+                            // eslint-disable-next-line react/no-danger
+                            dangerouslySetInnerHTML={{
+                                __html: arrowRight.markup,
+                            }}
+                        />
+                    </a>
+                </div>
+            </div>
+        );
+    }
+}
+
+class AccountCreationForm extends Component<
+    AccountCreationFormProps,
+    {
+        password: string,
+        isLoading: boolean,
+        isError: boolean,
+    }
+> {
+    onSubmit = (ev: Event) => {
+        ev.preventDefault();
+        this.setState({
+            isLoading: true,
+            isError: false,
+        });
+
+        reqwest({
+            url: '/password/guest',
+            method: 'post',
+            data: {
+                csrfToken: this.props.csrfToken,
+                token: this.props.accountToken,
+                password: this.state.password,
+            },
+            success: () => {
+                this.props.onAccountCreated();
+            },
+            error: response => {
+                try {
+                    const apiError = JSON.parse(response.responseText)[0];
+                    this.setState({
+                        isError: true,
+                        errorReason: apiError.description,
+                    });
+                } catch (exception) {
+                    this.setState({ isError: true });
+                }
+            },
+            complete: () => {
+                this.setState({ isLoading: false });
+            },
+        });
+    };
+
+    handlePasswordChange = (ev: Event) => {
+        if (!(ev.target instanceof HTMLInputElement)) {
+            return;
+        }
+        this.setState({ password: ev.target.value });
+    };
+
+    render() {
+        const { isError, errorReason } = this.state;
```

Comment:
`errorReason` isn't in the type declaration for the `AccountCreationForm` component's state.